monetio/sat/_tempo_l2_no2_mm.py

```diff
@@ -0,0 +1,246 @@
+"""TEMPO L2 NO2 data file reader.
+
+History:
+
+
+"""
+
+import logging
+import os
+import sys
+import warnings
+from collections import OrderedDict
+from glob import glob
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+from cftime import num2pydate
+from netCDF4 import Dataset
+
+
+def _open_one_dataset(fname, variable_dict):
+    """Read locally stored INTELSAT TEMPO NO2 level 2
+    Parameters
+    ----------
+    fname : str
+        Local path to netCDF4 (HDF5) file.
+    variable_dict : dict
+
+    Returns
+    -------
+    ds : xr.Dataset
+    """
+    print("reading " + fname)
+
+    ds = xr.Dataset()
+
+    dso = Dataset(fname, "r")
+    lon_var = dso.groups["geolocation"]["longitude"]
+    lat_var = dso.groups["geolocation"]["latitude"]
+    time_var = dso.groups["geolocation"]["time"]
+    time_units = time_var.units
+
+    ds["lon"] = (
+        ("x", "y"),
+        lon_var[:].squeeze(),
+        {"long_name": lon_var.long_name, "units": lon_var.units},
+    )
+    ds["lat"] = (
+        ("x", "y"),
+        lat_var[:].squeeze(),
+        {"long_name": lat_var.long_name, "units": lat_var.units},
+    )
+    ds["time"] = (
+        ("time",),
+        num2pydate(time_var[:].squeeze(), time_units),
+        {"long_name": time_var.long_name, "units": time_var.units},
```

Comment:
```suggestion
        {"long_name": time_var.long_name},
```
Units attr no longer relevant after decoding

Reply:
I think that the units attribute is worth keeping. However, the way it's now fails at save when it tries to encode it, because it is creating a DatetimeIndex with it's own encoding.
I'll find a way to fix it and commit again. Sorry for this.

Reply:
You could call it "original_units" instead?

Reply:
The units need to be the real units, that will be found in the attributes of the netcdf that gets created if someone saves it. The problems seems to lie in the fact that xarray is creating and index with time, not just the (expected) coordinates and variable. I need to find out a way to get rid of that.
Normally, with xarray, if you set valid units for time, it should be able to take care of applying the correct encoding on its own.

Reply:
OK, you were right and I was wrong about this. XArray manages the encoding on it's own, and even though I can force it (by setting an ds["time"].encoding = {"dtype":  "float32", "units": time_var.units} ) it doesn't seem to make a lot of sense. The only *nicer* thing about doing that is for people to might save de file and want to process it with other tools, that might need consistency in the time units.
Just going with your commit is the right way to go, and I just leave this here in case we need to refer to it in the future.